aten/src/ATen/core/PythonFallbackKernel.cpp

```diff
@@ -4,11 +4,21 @@
 
 namespace {
 
+// TLS saving the state of the include/exclude sets on entry to the dispatcher
+// This is set in the pythonTLSSnapshot fallback and used by the Python fallback.
+thread_local c10::optional<c10::impl::LocalDispatchKeySet> tls_on_entry;
+
 void pythonFallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
+  c10::impl::LocalDispatchKeySet keyset_on_dispatch_entry = c10::impl::tls_local_dispatch_key_set();
+
+  TORCH_INTERNAL_ASSERT(tls_on_entry.has_value());
+  c10::impl::_force_tls_local_dispatch_key_set(tls_on_entry.value());
+
   // If Python Mode is active, use its PyInterpreter for dispatch
   const auto& maybe_python_mode_state = at::impl::PythonModeTLS::get_state();
   if (maybe_python_mode_state) {
     maybe_python_mode_state->pyinterpreter()->dispatch(op, stack, maybe_python_mode_state);
+    c10::impl::_force_tls_local_dispatch_key_set(keyset_on_dispatch_entry);
```

Comment:
This should actually be a RAII guard or bad things are gonna happen. cc @zou3519 because one of those actually lives directly in functorch (an RAII guard struct for arbitrarily modifying include/exclude TLS), and we totally want a version of that in core.